spec/src/main/asciidoc/chapters/introduction/algebra.asciidoc

```diff
@@ -0,0 +1,334 @@
+The structure _s_ contains no other elements.
+
+For example, the query `from Book` might return a list containing structures like:
+
+[source]
+{
+  this: {
+    isbn: "9781857232738",
+    title: "Feersum Endjinn",
+    pages: 279,
+    year: 1994,
+    authors: [
+      {
+        name: "Iain M. Banks",
+        born: 'February 16, 1954',
+        died: 'June 9, 2013'
+      }
+    ]
+  }
+}
+
+And the query `from Book as book` returns a list containing structures like:
+
+[source]
+{
+  book: {
+    isbn: "9781857232738",
+    title: "Feersum Endjinn",
+    pages: 279,
+    year: 1994,
+    authors: [
+      {
+        name: "Iain M. Banks",
+        born: 'February 16, 1954',
+        died: 'June 9, 2013'
+      }
+    ]
+  }
+}
+
+[[joins]]
+=== Joins
+
+Every join has a _target_, which must be:
+
+- an entity,
+- a collection, or
+- an embeddable type.
+
+A join introduces a new labelled element to a result list.
+Evaluating a sequence of joins produces a result list in which each structure contains a
+labelled element corresponding to each join.
+The label of this element is often called the _alias_ or _identification variable_ of the join.
+
+==== Joins to named entities
+
+A join may specify the name of an entity:
+
+    [from], Y as y
+
+where `Y` is the name of the entity, `y` is the label (alias) assigned to it, and `[from]` is a root entity or a join.
+As before, the keyword `as` is optional.
+
+This kind of join produces a cartesian product.
+For each structure _r_ of the result list of the operation on which the join acts, and for each record _y_ of any entity type which inherits `Y`, the result list contains a structure _r'_ containing all labelled elements of _r_ together with the record _y_ labelled by the alias `y`.
+
+For example, this query is a cartesian product:
+
+    from Book as book, Loan as loan
+
+This query evaluates to a result list containing a structure for each pairing of a record of entity `Book` with a record of entity `Loan`.
+The structure contains two labelled elements, one for each entity, each labelled by its corresponding alias, `book`, and `loan`, respectively.
+The list might contain structures this:
+
+[source]
+{
+  book: {
+    isbn: "9781857232738",
+    title: "Feersum Endjinn",
+    pages: 279,
+    year: 1994,
+    authors: [
+      {
+        name: "Iain M. Banks",
+        born: 'February 16, 1954',
+        died: 'June 9, 2013'
+      }
+    ]
+  },
+  loan: {
+    bookIsbn: "9781932394153",
+    borrowerCard: "XYZ-123"
+  }
+}
+
+Note that there is no meaningful relationship between the `book` and the `loan`.
+
+A join to a named entity may be immediately followed by a <<restriction,restriction>>.
+In this case, the syntax is slightly different:
+
+    [from] join Y as y on [predicate]
+
+where `[predicate]` is a predicate, as defined later in <<predicates>>.
+
+For example:
+
+    from Book as book
+    join Loan as loan
```

Comment:
This `join`  section might bring a good discussion.

For example, in NoSQL with MongoDB it will be just a subdocument, thus, I can skip this join.

It is fair because we want to keep compatibility.


But if we could abstract it would be amazing, letting the databases target do their job.